src/main/java/com/zerobase/zbfintech/service/UserSignUpService.java

```diff
@@ -0,0 +1,93 @@
+package com.zerobase.zbfintech.service;
+
+import com.zerobase.zbfintech.dto.SignupForm;
+import com.zerobase.zbfintech.entity.User;
+import com.zerobase.zbfintech.exception.CustomException;
+import com.zerobase.zbfintech.repository.UserRepository;
+import org.springframework.stereotype.Service;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.Locale;
+import java.util.Optional;
+
+import static com.zerobase.zbfintech.exception.ErrorCode.*;
+
+@Service
+public class UserSignUpService {
+
+    private final UserRepository userRepository;
+    private final EmailService emailService;
+
+    public UserSignUpService(UserRepository userRepository, EmailService emailService) {
+        this.userRepository = userRepository;
+        this.emailService = emailService;
+    }
+
+    @Transactional
+    public String userSignUp(SignupForm form) {
+        if (isEmailExists(form.getEmail())) {
+            throw new CustomException(ALREADY_REGISTER_USER);
+        } else {
+            User user = signup(form);
+            LocalDateTime now = LocalDateTime.now();
+            String code = getRandomCode();
+            emailService.sendEmail(form.getEmail(),
+                    "Verification Email!",
+                    getVerificationEmailBody(user.getEmail(), user.getUsername(), code));
+            changeCustomerValidateEmail(user.getId(), code);
+            return "회원 가입에 성공했습니다.";
+        }
+    }
+
+    public User signup(SignupForm form) {
+        return userRepository.save(User.from(form));
+    }
+
+    private boolean isEmailExists(String email) {
+        return userRepository.findByEmail(email.toLowerCase(Locale.ROOT)).isPresent();
+    }
+
+    private void changeCustomerValidateEmail(Long userId, String verificationCode) {
+        Optional<User> userOptional = userRepository.findById(userId);
+
+        if(userOptional.isPresent()) {
+            User user = userOptional.get();
+            user.setVerificationCode(verificationCode);
+            user.setVerifyExpiredAt(LocalDateTime.now().plusDays(1));
+        }else {
+            throw new CustomException(ALREADY_REGISTER_USER);
+        }
```

Comment:
optional 에서 제공하는 메서드 체이닝 로직을 이용하여 orElseThrow 를 통해 조금 더 간결하게 작성해보는 것은 어떨까요?

https://velog.io/@wonizizi99/Optional%EC%9D%98-orElseorElseThrow-%EC%82%AC%EC%9A%A9%EB%B2%95

```suggestion
       User user = userRepository.findById(userId).orElseThrow(() -> new CustomException(ALREADY_REGISTER_USER));
       user.setVerificationCode(verificationCode);
       user.setVerifyExpiredAt(LocalDateTime.now().plusDays(1));
```